com.weibellab.comms/Runtime/ReliableCommunication.cs

```diff
@@ -307,6 +319,123 @@ public void StartConnection()
             }
         }
 
+        /// <summary>
+        /// Sets the target IP and Port based on set strategy
+        /// By default, the user manually sets the port and IP.
+        /// Optionally, they can ping a web server to get other device info.
+        /// </summary>
+        private async void setTargetAndStartConnection()
+        {
+            // yield return null;
+            if (Application.isPlaying) {
+                Task<bool> task;
+                switch (this.strategy)
+                {
+                    case TargettingStrategy.Manual:
+                        this.StartConnection();
+                        break;
+                    case TargettingStrategy.Web:
+                        task = this.getAddressFromWeb();
+                        await task;
+                        while (task.Result == false && Application.isPlaying) {
+                            Debug.Log(LogName + "Failed to Connect");
+                            Task.Delay(1000).Wait();
+                            task = this.getAddressFromWeb();
+                            await task;
+                        }
+                        this.StartConnection();
+                        break;
+                    case TargettingStrategy.UDP:
+                        task = this.getAddressFromUdp();
+                        await task;
+                        while (task.Result == false && Application.isPlaying) {
+                            Debug.Log(LogName + "Failed to Connect");
+                            Task.Delay(100).Wait();
+                            task = this.getAddressFromUdp();
+                            await task;
+                        }
+                        this.StartConnection();
+                        break;
+                    default:
+                        throw new Exception(LogName + "Unknown Connection Strategy");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the target Port and IP from a web request.
+        /// POSTs data about this Reliable Communication
+        /// gets a response with server/client address(es)
+        /// </summary>
+        private async Task<bool> getAddressFromWeb()
+        {
```

Comment:
If Client:
POST data about self to webserver. Webserver will respond with the server's port and IP.

If Server:
POST data about self to webserver which will create the room and set the passkey. If room already exists, update the server's IP and return a list of clients in the room.